google_scraper.py

```diff
@@ -55,22 +55,26 @@ def search(url):
             element.send_keys(Keys.PAGE_DOWN)
             time.sleep(0.3)  # bot id protection
 
-    print("[%] Reached end of Page.")
+    print("==> Reached end of page")
 
     time.sleep(1)
     # Get page source and close the browser
+    print("==> Saving page source...")
     source = browser.page_source
-    f = open('dataset/logs/google/source.html', 'w+')
-    f.write(source)
+    with open('dataset/logs/google/source.html', 'w+', encoding='utf-8', errors='replace') as f:
```

Comment:
Works for Python 3. Python 2 Solution can be implemented using the `io` module.

Python 2: https://docs.python.org/2/library/io.html
Python 3: https://docs.python.org/3/library/io.html